### PR TITLE
split the error_code.go file into three files

### PR DESCRIPTION
**Pull request by gregwebs**
## What have you changed? (required)

split the error_code.go file into three files

This is purely for code organization (readability) as suggested by @disksing. I think it turns out to be fairly easy to split things up in error_code because there are sizeable independent chunks. They are all pulled back together by `NewJSONFormat`, but I think that is okay.

## What are the type of the changes (required)?

No change to existing code. Just moving some code to a different file of the same package, so its all the same to the go compiler.

## How has this PR been tested (required)?

`make && make check && go test pkg/error_code/error_code.go`

### CODE REVIEW

File: pkg/error_code/operation.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+// AddOp is constructed by Op. It allows method chaining with AddTo.
+type AddOp func(ErrorCode) OpErrCode
+
+// AddTo adds the operation from Op to the ErrorCode
+func (addOp AddOp) AddTo(err ErrorCode) OpErrCode {
+	return addOp(err)
+}
+
+/*
+Op adds an operation to an ErrorCode with AddTo.
```
Comment (siddontang) [inline pkg/error_code/operation.go]: I prefer using `//` all the time.

Comment (siddontang) [inline pkg/error_code/operation.go]: I prefer using `//` all the time.

Comment (gregwebs) [reply pkg/error_code/operation.go]: okay, done now.

**Comment by disksing** [comment]
PTAL @siddontang

**Pull request merged**